src/rnpkeys/rnpkeys.cpp

```diff
@@ -254,6 +254,10 @@ rnp_cmd(rnp_cfg_t *cfg, cli_rnp_t *rnp, optdefs_t cmd, const char *f)
     case CMD_GENERATE_KEY: {
         if (f == NULL) {
             f = rnp_cfg_getstr(cfg, CFG_USERID);
+            if(rnp_cfg_getstr(cfg, CFG_USERID, 2) != NULL) {
+                fprintf(stderr, "only userid is supported for generated keys\n");
```

Comment:
The idea of this PR was to update code to be able to generate keys with multiple user id's, not just report an error. While key generation function supports single key id at the moment, you still may use `rnp_key_add_uid()` on the generated key. Also it worth adding `rnp_op_generate_add_uid()` method, so user id would be added in a single pass.